source/github-helpers/get-default-branch.ts

```diff
@@ -14,6 +18,9 @@ async function fromDOM(): Promise<string | undefined> {
 		return undefined;
 	}
 
+	await domLoaded; // DOM-based filter
+	await delay(100);
```

Comment:
`delay` is generally a last resort, but `domLoaded` should not be needed in the first place, we already have `elementReady`.

Reply:
Yeah, I know that elementReady means same works.
But the elementReady or domLoaded not works exactly in this case.

Reply:
We can resolve this with this change on `element-ready`
https://github.com/sindresorhus/element-ready/blob/036c1085b10ef01313ec60ef183f399dfdd05322/index.js#L7-L8C34

```js
// AS-IS
const isDomReady = target =>
	['interactive', 'complete'].includes((target.ownerDocument || target).readyState);

// TO-BE
const isDomReady = target =>
	['complete'].includes((target.ownerDocument || target).readyState);
```

Reply:
There's a misunderstanding in how these two work. If you wait for the whole document to be ready, then you don't need elementReady, because at that point it behaves as a simple `querySelector`/`select`. If the element is not part of the original document, then neither dom-loaded nor element-ready should be used. 

If you're looking for an element/component that's loaded dynamically, it's best to use the selector observer. Or maybe in this case it's easier to use `elementReady(selector, {stopOnDomLoaded: false, timeout: 1000})`

Reply:
We need to refactor this API 🤔

Reply:
I fixed just returning `undefined` when elementReady failed to find element